projects/ccf-rui/src/app/shared/components/labeled-slide-toggle/labeled-slide-toggle.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, Input, Output, EventEmitter, HostBinding } from '@angular/core';
+
+/**
+ * Generic toggle slider component
+ */
+@Component({
+  selector: 'ccf-labeled-slide-toggle',
+  templateUrl: './labeled-slide-toggle.component.html',
+  styleUrls: ['./labeled-slide-toggle.component.scss']
+})
+
+export class LabeledSlideToggleComponent {
+
+  @HostBinding('class') readonly clsName = 'ccf-labeled-slide-toggle';
+
+  /**
+   * Whether or not the slider is disabled
+   */
+  @Input() slideDisabled = false;
+
+  /**
+   * The category of data selected
+   */
+  @Input() dataType: string;
+
+  /**
+   * The two selection options to be toggled
+   */
+  @Input() labels: string[];
+
+  /**
+   * Determines if left toggle option is selected
+   */
+  left = true;
+
+  /**
+   * Emits the datatype with the currently selected option
+   */
+  @Output() toggleChanged = new EventEmitter<{}>();
```

Comment:
Output type will be `string` when the other changes are implemented

Reply:
This should also ideally be `@Input() value: string` and `@Input() valueChange...` to be consistent with other pure components

Reply:
The easiest solution without involving lifecycle hooks would then be something like
```
@Input() set value(label: string) {
  this.left = !this.labels || label !== this.labels[1];
}
```
And if we want to store the value then
```
class LabeledSliderToggle {
  ...

  @Input() set value() ...
  get value(): string { return this._value; }
  private _value: string;
```

Reply:
What is the valueChange input? I thought the input value was all we needed.

Reply:
It is supposed to be `@Output() valueChange ...`

Reply:
Ahh ok